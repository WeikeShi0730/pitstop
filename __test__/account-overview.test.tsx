import { render, RenderResult } from "@testing-library/react";
import AccountOverview from "../components/account-overview.component";

let documentBody: RenderResult;
describe("<AccountOverview />", () => {
  beforeEach(() => {
    documentBody = render(
      <AccountOverview
        currentUser={null}
        orderHistoryItems={[]}
        setCurrentSelection={() => {}}
      />
    );
  });
  it("shows content in <AccountOverview />", () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
