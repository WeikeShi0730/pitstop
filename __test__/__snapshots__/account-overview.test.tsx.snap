// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountOverview /> shows content in <AccountOverview /> 1`] = `
<body>
  <div>
    <div
      class="m-10 flex flex-col w-full gap-y-5 text-sm md:text-base"
    >
      <div
        class="w-fit flex justify-start text-slate-700 text-lg md:text-2xl py-1 border-b border-slate-700"
      >
        Overview
      </div>
      <div
        class="my-5 text-xl md:text-3xl flex justify-center"
      >
        Hi, 
        !
      </div>
      <div
        class="w-full max-w-xs sm:max-w-sm md:max-w-md lg:max-w-lg m-auto"
      >
        <div
          class="text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg py-5 px-5 shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200"
        >
          <div
            class="flex flex-col items-center"
          >
            <div
              class="text-center self-start m-3 text-lg md:text-xl"
            >
              Summary
            </div>
            <div
              class="m-3"
            >
              <div
                class="text-center py-2 px-4 leading-8"
              >
                You've bought
                 
                <span
                  class="underline-primary font-normal text-lg md:text-xl"
                >
                  0
                </span>
                 
                stickers!
              </div>
              <div
                class="text-center py-2 px-4 leading-8"
              >
                CAD
                 
                <span
                  class="underline-primary font-normal text-lg md:text-xl"
                >
                  0.00
                </span>
                 
                in total. ðŸ’¸
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full max-w-xs sm:max-w-sm md:max-w-md lg:max-w-lg m-auto"
      >
        <div
          class="text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg py-5 px-5 shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200"
        >
          <div
            class="flex flex-col items-center"
          >
            <div
              class="text-center self-start m-3 text-lg md:text-xl"
            >
              Latest Order
            </div>
            <div
              class="m-3 text-center"
            >
              No orders yet.
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full max-w-xs sm:max-w-sm md:max-w-md lg:max-w-lg m-auto"
      >
        <div
          class="text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg py-5 px-5 shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200"
        >
          <div
            class="flex flex-col items-center"
          >
            <div
              class="text-center self-start m-3 text-lg md:text-xl"
            >
              Update my password?
            </div>
            <button
              class="text-sm md:text-base m-3 bg-slate-700 py-2 px-4 text-slate-100 rounded outline-none font-light hover:shadow-md hover:shadow-slate-700 hover:bg-slate-500 hover:text-white transition-all ease-in-out duration-200"
            >
              Send a password update email
            </button>
          </div>
        </div>
      </div>
      <div
        class="w-full max-w-xs sm:max-w-sm md:max-w-md lg:max-w-lg m-auto"
      >
        <div
          class="text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg py-5 px-5 shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200"
        >
          <div
            class="flex flex-col items-center"
          >
            <div
              class="text-center self-start m-3 text-lg md:text-xl"
            >
              Need support?
            </div>
            <a
              class="text-sm md:text-base m-3 bg-slate-700 py-2 px-4 text-slate-100 rounded outline-none font-light hover:shadow-md hover:shadow-slate-700 hover:bg-slate-500 hover:text-white transition-all ease-in-out duration-200"
              href="/contact"
            >
              Contact us
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
